---
title: "SweeD results"
author: "Markus Stetter"
date: "10/30/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(root.dir ='~/Documents/Science/hohenheim_phd/amaranth_domestication/')
```

```{r,eval=FALSE}
setwd('~/Documents/Science/hohenheim_phd/amaranth_domestication/code/sweed/')
```
## Selective sweeps

```{r}
library(data.table)
library(tidyverse)
library(cowplot)
library(qqman)
library(UpSetR)
library(wesanderson)
library(gggenes)
```

## Including Plots

You can also embed plots, for example:
```{r}

plot_all <- function(data,header){
cut <- quantile(data$Likelihood,0.995)

ggplot(data,aes(Position/1e6,Likelihood,color=as.factor(CHR)))+
  facet_grid(.~CHR,space = 'free_x',scales='free_x') + 
#  scale_fill_manual(values = alpha(rep(c('white','grey90'),8),alpha=0.1))+
#  geom_rect(aes(fill = as.factor(CHROM)),xmin = -Inf,xmax = Inf, ymin = -Inf,ymax = Inf) +
  geom_point(size=1) +
  labs(x= 'Position (Mb)',y= "Likelihood") +
  scale_x_continuous(breaks = c(0,10,20,30))+
  geom_hline(data=data.frame(yint=cut),aes(yintercept =yint,linetype ='dashed',color=alpha('red',0.6)))+
  geom_vline(data=data.frame(xint=c(14242288/1e6,14858076/1e6,5875169/1e6,	5975848/1e6),CHR=c('9','9','3','3')),
             aes(xintercept=xint),color='grey80',linetype='dashed')+
  theme(panel.spacing.x=unit(0.25, "lines")) +
    scale_color_manual(values =wes_palette("Darjeeling1",17, type = "continuous"))+
  theme( strip.background = element_rect(fill = alpha('lightblue',0.2)),strip.text = element_text(size=18)) +
  theme(legend.position="none",axis.text.y = element_text(size=18),axis.title = element_text(size=22))
} 
```



```{r}
getwd()
read_sweed_report <- function(set){
out_list <- list()
for (i in c(1:16)){
chr_data <- fread(paste('../../data/sweed/SweeD_Report.',set,'_',i,sep=''),skip = 1)
chr_data$CHR <- i
out_list[[i]] <- chr_data
}
dt <- rbindlist( out_list )
dt 
}
```

```{r}
caudatus <- read_sweed_report('caudatus_samples')
plot_all(caudatus,'A. caudatus')
ggsave('../../figures/sweed_results/caudatus.png',width = 16)
```
```{r}
cruentus <- read_sweed_report('cruentus_samples_corrected')
plot_all(cruentus,'A. cruentus')
ggsave('../../figures/sweed_results/cruentus.png',width = 16)

```
```{r}
hypochondriacus <- read_sweed_report('hypochondriacus_samples_corrected')
plot_all(hypochondriacus,'A. hypochondriacus')
ggsave('../../figures/sweed_results/hypochondriacus.png',width = 16)

```

```{r}
hybridus <- read_sweed_report('hybridus_samples')
plot_all(hybridus,'A. hybridus')
ggsave('../../figures/sweed_results/hybridus.png',width = 16)

```
```{r}
quitensis <- read_sweed_report('quitensis_samples')
plot_all(quitensis,'A. quitensis')
ggsave('../../figures/sweed_results/quitensis.png',width = 16)

```

```{r}
wild_SA <- read_sweed_report('wild_SA')
plot_all(wild_SA,'Wild South America')
ggsave('../../figures/sweed_results/wild_SA.png',width = 16)

plot_all(wild_SA,'Wild South America')
```




```{r}
wild_combined <- read_sweed_report('wild_combined')
plot_all(wild_combined,'All wild')
ggsave('../../figures/sweed_results/wild_combined.png',width = 16)

```

```{r}
cut <- quantile(wild_SA$Likelihood,0.995)
ggplot(filter(wild_SA,CHR==3,Position<6.3e6 &Position>5.7e6),aes(Position/1e6,Likelihood,color=as.factor(CHR)))+
#  scale_fill_manual(values = alpha(rep(c('white','grey90'),8),alpha=0.1))+
#  geom_rect(aes(fill = as.factor(CHROM)),xmin = -Inf,xmax = Inf, ymin = -Inf,ymax = Inf) +
  geom_point(size=2) +
  labs(x= 'Position (Mb)',y= "Likelihood",title='Chromosome 3') +
  geom_hline(data=data.frame(yint=cut),aes(yintercept =yint,linetype ='dashed',color=alpha('red',0.6)))+
  geom_vline(data=data.frame(xint=c(5875169/1e6,	5975848/1e6),CHR=c('3','3')),
             aes(xintercept=xint),color='grey80',linetype='dashed')+
  theme(panel.spacing.x=unit(0.25, "lines")) +
    scale_color_manual(values =wes_palette("Darjeeling1",17, type = "continuous"))+
  theme( strip.background = element_rect(fill = alpha('lightblue',0.2)),strip.text = element_text(size=18)) +
  theme(legend.position="none",axis.text.y = element_text(size=18),axis.title = element_text(size=22))

ggsave('../../figures/sweed_results/chr3_wild_SA.png',height=6)
```




## Hybridus sig window on Chr 3

```{r}
cut <- quantile(wild_SA$Likelihood,0.99)
data1 <- wild_SA%>%
  mutate(rno = 1:length(CHR),sig=(Likelihood>cut))%>%
  filter(sig==TRUE,CHR==3)
data1
```
## Sweed plots combined

```{r,fig.height=12,fig.width=12}
no_axis <-    theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),plot.margin = unit(c(.5,0.1,0.1,0.5), "cm"), 
        axis.ticks.length = unit(0, "mm"))
middle <- theme(strip.background = element_blank(),strip.text = element_blank())
plot_grid(
plot_all(hybridus,'A. hybridus')+no_axis,
plot_all(quitensis,'A. quitensis')+no_axis+middle,
plot_all(caudatus,'A. caudatus')+no_axis+middle,
plot_all(cruentus,'A. cruentus')+no_axis+middle,
plot_all(hypochondriacus,'A. hypochondriacus')+middle+theme(plot.margin = unit(c(0.5,0.1,0.1,0.5), "cm")),
ncol = 1,align = 'v',rel_heights = c(100,95,95,95,110),labels = c("A","B","C","D","E"),label_size = 22
)
ggsave('../../figures/sweed_results/combined_sweed_results.pdf',height=16,width = 16)
ggsave('../../figures/sweed_results/combined_sweed_results.png',height=16,width = 16)

```

```{r}
get_sig_windows <- function(data){
cut <- quantile(data$Likelihood,0.995)
data1 <- data%>%
  mutate(rno = 1:length(CHR),sig=(Likelihood>cut))%>%
  filter(sig==TRUE) %>%
  select(rno)
data1$rno
}
caudatus1 <- get_sig_windows(caudatus)
cruentus1 <- get_sig_windows(cruentus)
hybridus1 <- get_sig_windows(hybridus)
hypochondriacus1 <- get_sig_windows(hypochondriacus)
quitensis1 <- get_sig_windows(quitensis)
wildSA1 <- get_sig_windows(wild_SA)
wildall1 <- get_sig_windows(wild_combined)

metadata <- data.frame(set=c('caudatus','cruentus','hypochondriacus','hybridus','quitensis'),
                       species=c('caudatus','cruentus','hypochondriacus','hybridus','quitensis')
                       )
mycolor=c('#33a02c','#1f78b4','#a6cee3','#e31a1c','#ff7f00','#fdbf6f','#fb9a99','#b2df8a','#cab2d6','#6a3d9a')

{pdf('../../figures/sweed_results/overlapping_sweeps.pdf',onefile = FALSE,width = 12)

upset(fromList(list(caudatus=caudatus1,
                    cruentus=cruentus1,
                    hypochondriacus=hypochondriacus1,
                    hybridus=hybridus1,
                    quitensis=quitensis1)), 
      order.by = "freq",
      sets.x.label = 'Sweeps per group',
      text.scale = c(2, 2,1, 1, 2, 2), 
      set.metadata = list(data = metadata,plots = list(list(type = "matrix_rows",  assign = 10,
    column = "species", colors = c(caudatus="#33a02c", cruentus="#1f78b4", hypochondriacus = "#ff7f00",hybridus='#e31a1c',quitensis='#fb9a99'), alpha = 1))))
dev.off()}



upset(fromList(list(caudatus=caudatus1,
                    cruentus=cruentus1,
                    hypochondriacus=hypochondriacus1,
                    hybridus=hybridus1,
                    quitensis=quitensis1)), 
      order.by = "freq",
      sets.x.label = 'Sweeps per group',
      text.scale = c(1.3, 1.3, 3, 1, 2, 0.75), mb.ratio = c(0.45, 0.45))
dev.off()

#x = # of genes in common between two groups.
#n = # of genes in group 1.
#D = # of genes in group 2.
#N = total genes, in this case the 17611 genes with good spots on the Kim lab full genome chips.
#C(a,b) is the number of combinations of a things taken b at a time.


#C(D, x) * C(N-D, n-x) / C(N,n)



countOverlap<-function(total=19791, numgA=99, numgB=99, replace=TRUE){
    groupA<-sample.int(total, numgA, replace=replace)
    groupB<-sample.int(total, numgB, replace=replace)
    return(length(intersect(groupA, groupB)))
}

tmpres<-replicate(1000000, countOverlap(total=19791, numgA=99, numgB=99))

#if your true observed value was 17:
pval<-mean(tmpres >= 4)
pval

```


