---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{bash}
#!/bin/bash

STATS_FILE=confirmation_quitensis.thetasWindow.gz.pestPG
POPULATION=quitensis

awk '{print $1, $2, $4, $5, $9, $14}' $STATS_FILE |  sed '1d;$d' | sed 's/(/\t/g' | sed 's/)/\t/g' | sed 's/,/\t/g' | awk '{print $7, $5, $6, $8, $9, $10, $11}' | sed 's/ /\t/g' | sort -k1,1 -k2,2n > $POPULATION_tests.bedgraph
```



##### packages and necessary libraries
```{r}
library(data.table)
library(RcppCNPy)
library("factoextra")
library('ggfortify')
library('cowplot')
library('dplyr')
nice_layout<-   theme_cowplot()+
    panel_border(color = "grey85", size = 1, linetype = 1,
  remove = FALSE, "black") 
```




#### load neutrality tests bedgrtaph file for per site calculation of Watterson and nucleotide divcersity

```{r}
  popualation_tests<-fread('/Users/josedias/mount/scratch/jgoncal1/outputs/caudatus_tests.bedgraph') %>%
          dplyr::rename(CHR=V1, StartPos=V2, EndPos=V3, tW=V4, tP=V5, Tajima=V6, nSites=V7) %>%
  mutate(Watterson=tW/nSites, pi=tP/nSites, species = "caudatus")
  
popualation_tests %>%
  select(CHR, StartPos, EndPos, pi) %>%
  fwrite("population_pi.bedgraph")

popualation_tests %>%
  select(CHR, StartPos, EndPos, tW) %>%
  fwrite("population_tW.bedgraph")


popualation_tests %>%
  select(CHR, StartPos, EndPos, Tajima) %>%
  fwrite("population_Tajima.bedgraph")


```


#### Convert bedgrpah R output to BigWig for adding quantitative trakcs on the Genome Browser

```{bash}
fa2size reference.fasta reference.sizes -detailed
bedGraphToBigWig in.bedGraph chrom.sizes out.bw
```


#### plot Watterson, pi and Tajima D

```{r}
all_species.filtered <- full_join(caudatus_tests, cruentus_tests) %>%
  full_join(quitensis_tests) %>%
  full_join(hybridus_tests) %>%
  full_join(hypochondriacus_tests) %>%
  full_join(quitensis_tests) %>%
    filter(nSites>250)


all_species.filtered %>%
    group_by(species) %>%
summarise_if(is.numeric, mean)



    

  plot_Tajima_all_species_filtered<-            ggplot(all_species.filtered, aes(x=Tajima)) +
geom_density(aes(color = species, fill=species), alpha=0.4) +
nice_layout +
           labs(title = "Tajima")
      
      
             plot_pi_all_species_filtered<- ggplot(all_species.filtered, aes(x=pi)) +
geom_density(aes(color = species, fill=species), alpha=0.4) +
nice_layout +
        labs(title = "pi")

                   
             plot_Watterson_all_species_filtered<-     ggplot(all_species.filtered, aes(x=Watterson)) +
geom_density(aes(color = species, fill=species), alpha=0.4) +
nice_layout      +     
           labs(title = "Watterson")
           
         
           
title.filtered <- ggdraw() + 
draw_label(
"All species nsites >250",
fontface = 'bold',
x = 0,
hjust = 0
) 

plot_grid(plot_Tajima_all_species_filtered, plot_pi_all_species_filtered, plot_Watterson_all_species_filtered, labels="AUTO", title.filtered)

```